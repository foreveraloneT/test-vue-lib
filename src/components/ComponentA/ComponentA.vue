<script setup lang="ts">
import { useI18n } from 'vue-i18n';

defineOptions({
  name: 'ComponentA',
});

const { t } = useI18n();
</script>

<template>
  <div class="tv-p-4 tv-rounded-lg tv-bg-green-500 tv-w-fit">
    {{ t('oapluscomponent.componenta.title') }}
  </div>
</template>
