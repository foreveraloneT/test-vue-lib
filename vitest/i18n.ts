import { config } from '@vue/test-utils';
import { createI18n } from 'vue-i18n';

import {
  en as messagesEN,
  th as messagesTH,
  ja as messagesJA,
  ko as messagesKO,
  zh_TW as messagesZhTW,
} from '../src/locale';

function hasAllTranslation(key: string): boolean {
  return key in messagesEN
    && key in messagesTH
    && key in messagesJA
    && key in messagesKO
    && key in messagesZhTW;
}

const msg = Object.keys(messagesEN)
  .filter((key) => hasAllTranslation(key))
  .reduce((acc, key) => ({
    ...acc,
    [key]: key,
  }), {});

const i18n = createI18n({
  locale: 'en',
  messages: {
    en: msg,
  },
  legacy: false,
  globalInjection: true,
  missing(_, key) {
    throw new Error(`unable to find key: ${key} in some locale`);
  },
});

config.global.plugins.push(i18n);
