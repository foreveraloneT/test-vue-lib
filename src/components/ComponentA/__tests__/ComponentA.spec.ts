import { shallowMount } from '@vue/test-utils';
import { describe, it, expect } from 'vitest';

import ComponentA from '../ComponentA.vue';

describe('ComponentA', () => {
  describe('snapshots', () => {
    it('should render properly', () => {
      const wrapper = shallowMount(ComponentA);

      expect(wrapper.element).matchSnapshot();
    });
  });
});
