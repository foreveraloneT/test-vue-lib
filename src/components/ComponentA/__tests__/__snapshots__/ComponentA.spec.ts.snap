// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ComponentA > snapshots > should render properly 1`] = `
<div
  class="tv-p-4 tv-rounded-lg tv-bg-green-500 tv-w-fit"
>
  tvcomponent.componenta.title
</div>
`;
